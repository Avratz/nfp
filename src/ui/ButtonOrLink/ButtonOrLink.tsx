'use client'
import { Link } from '@/ui/Link/Link'
import { ComponentProps, forwardRef } from 'react'

export type ButtonOrLinkProps = ComponentProps<'button'> &
  Omit<ComponentProps<typeof Link>, 'href'> & {
    href?: string
  }

export const ButtonOrLink = forwardRef<
  HTMLButtonElement & HTMLAnchorElement,
  ButtonOrLinkProps
>(function ButtonOrLinkForward({ href, children, ...props }, ref) {
  if (typeof href !== 'undefined') {
    return (
      <Link href={href} {...props} ref={ref}>
        {children}
      </Link>
    )
  }

  return (
    <button {...props} ref={ref}>
      {children}
    </button>
  )
})
