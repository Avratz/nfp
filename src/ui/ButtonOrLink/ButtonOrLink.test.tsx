import { render } from '@testing-library/react'
import { ButtonOrLink } from './ButtonOrLink'

describe('ButtonOrLink', () => {
  it('renders default', () => {
    const { container } = render(<ButtonOrLink>link</ButtonOrLink>)
    expect(container).toMatchSnapshot()
  })
})
