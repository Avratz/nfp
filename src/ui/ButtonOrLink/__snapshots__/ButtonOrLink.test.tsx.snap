// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonOrLink renders default 1`] = `
<div>
  <button>
    link
  </button>
</div>
`;
